client/actions.js

```diff
@@ -0,0 +1,308 @@
+    dispatch(idSearch());
+
+    fetch(`/viewemployee/${employeeId}`, {
+      headers: {'Content-Type': 'application/json'}
+    })
+      .then(response => response.json())
+      .then(response => {
+        dispatch(editForm(response));
+        document.getElementById('employee-id').value = response.id;
+        document.getElementById('employee-first').value = response.first_name;
+        document.getElementById('employee-last').value = response.last_name;
+        document.getElementById('employee-photo').value = response.photo;
+        document.getElementById('employee-title').value = response.job_title;
+        document.getElementById('employee-description').value = response.job_description;
+        document.getElementById('employee-email').value = response.email;
+        document.getElementById('employee-manager').value = response.manager_id;
+        dispatch(readyToEdit());
+      })
+  }
+}
+
+const editSubmitted = () => {
+  return { type: 'EDIT_SUBMITTED' }
+}
+
+const editSaved = response => {
+  return { type: 'EDIT_SAVED', response }
+}
+
+const saveUpdate = employee => {
+  return dispatch => {
+    dispatch(editSubmitted());
+    fetch('/updateemployee/', {
+      method: 'PUT',
+      headers: {'Content-Type': 'application/json'},
+      body: JSON.stringify(employee)
+    })
+      .then(response => response.json())
+      .then(response => {
+          dispatch(editSaved(response))
+      })
+  }
+}
+
+const deleteSubmitted = () => {
+  return { type: 'DELETE_EMPLOYEE' }
+}
+
+const deleteConfirmed = () => {
+  return { type: 'DELETE_CONFIRMED' }
+}
+
+const deleteError = () => {
+  return { type: 'DELETE_ERROR' }
+}
+
+const employeeDeleted = () => {
+  return { type: 'EMPLOYEE_DELETED' }
+}
+
+const deleteProfile = employee => {
+  return dispatch => {
+    dispatch(deleteSubmitted());
+    const confirm = window.confirm(`Are you sure you would like to delete Employee ${employee.id} ${employee.first} ${employee.last}?`);
+    if (!confirm) {
+      return;
+    }
+    dispatch(deleteConfirmed());
+    fetch(`/deleteemployee/${employee.id}`, {
+      method: 'DELETE',
+      headers: {'Content-Type': 'application/json'}
+    })
+      .then(response => response.json())
+      .then(response => {
+        if (response.error) {
+          alert(response.error);
+          dispatch(deleteError());
+        }
+        else if (response.success) {
+          alert(response.success);
+          dispatch(employeeDeleted(response))
+        }
+      })
+  }
+}
+
+const orgSubmitted = () => {
+  return { type: 'ORG_SUBMITTED' }
+}
+
+const renderManager = response => {
+  return { type: 'RENDER_MANAGER', response }
+}
+
+const renderEmployee = response => {
+  return { type: 'RENDER_EMPLOYEE', response}
+}
+
+const renderPeers = response => {
+  return { type: 'RENDER_PEERS', response}
+}
+
+const renderReports = response => {
+  return { type: 'RENDER_REPORTS', response}
+}
+
+const renderOrgChart = employee => {
```

Comment:
Is there any reason your server can't just send the data the client needs instead of demanding multiple requests?